Mathlib/FieldTheory/GalConj.lean

```diff
@@ -0,0 +1,394 @@
+
+@[simp]
+theorem out_equiv_out {c₁ c₂ : GalConjClasses F E} : (c₁.out ≈g[F] c₂.out) ↔ c₁ = c₂ :=
+  @Quotient.out_equiv_out _ _ c₁ c₂
+#align gal_conj_classes.out_equiv_out GalConjClasses.out_equiv_out
+
+theorem equiv_zero_iff (x : E) : (x ≈g[F] 0) ↔ x = 0 := by
+  refine' ⟨fun h => _, fun h => by rw [h]⟩
+  cases' h with a ha
+  simp_rw [← ha, AlgEquiv.smul_def, map_zero]
+#align gal_conj_classes.equiv_zero_iff GalConjClasses.equiv_zero_iff
+
+theorem out_eq_zero_iff (c : GalConjClasses F E) : c.out = 0 ↔ c = 0 := by
+  rw [zero_def, eq_mk_iff_out, equiv_zero_iff]
+#align gal_conj_classes.out_eq_zero_iff GalConjClasses.out_eq_zero_iff
+
+theorem zero_out : (0 : GalConjClasses F E).out = 0 :=
+  (out_eq_zero_iff 0).mpr rfl
+#align gal_conj_classes.zero_out GalConjClasses.zero_out
+
+theorem mk_eq_zero_iff (x : E) : mk F x = 0 ↔ x = 0 := by
+  rw [mk_eq_iff_out, zero_out, equiv_zero_iff]
+#align gal_conj_classes.mk_eq_zero_iff GalConjClasses.mk_eq_zero_iff
+
+theorem mk_zero : mk F (0 : E) = 0 :=
+  (mk_eq_zero_iff 0).mpr rfl
+#align gal_conj_classes.mk_zero GalConjClasses.mk_zero
+
+nonrec def orbit (c : GalConjClasses F E) : Set E :=
+  c.orbit
+#align gal_conj_classes.orbit GalConjClasses.orbit
+
+instance [DecidableEq E] [Fintype (E ≃ₐ[F] E)] (c : GalConjClasses F E) : Fintype c.orbit :=
+  Quotient.recOnSubsingleton' c fun _ => Set.fintypeRange _
+
+theorem mem_orbit {x : E} {c : GalConjClasses F E} : x ∈ c.orbit ↔ mk F x = c :=
+  MulAction.orbitRel.Quotient.mem_orbit
+#align gal_conj_classes.mem_orbit GalConjClasses.mem_orbit
+
+theorem orbit_zero : (0 : GalConjClasses F E).orbit = {0} := by
+  ext; rw [mem_orbit, mk_eq_zero_iff, Set.mem_singleton_iff]
+#align gal_conj_classes.orbit_zero GalConjClasses.orbit_zero
+
+instance : Neg (GalConjClasses F E) :=
+  ⟨Quotient.lift (fun x : E => mk F (-x))
+      (by
+        rintro _ y ⟨f, rfl⟩; rw [eq]
+        use f; change f (-y) = -f y; rw [AlgEquiv.map_neg])⟩
+
+theorem mk_neg (x : E) : mk F (-x) = -mk F x :=
+  rfl
+#align gal_conj_classes.mk_neg GalConjClasses.mk_neg
+
+instance : InvolutiveNeg (GalConjClasses F E) :=
+  { (inferInstance : Neg (GalConjClasses F E)) with
+    neg_neg := fun x => by rw [← out_eq x, ← mk_neg, ← mk_neg, neg_neg] }
+
+theorem exist_mem_orbit_add_eq_zero (x y : GalConjClasses F E) :
+    (∃ a b : E, (a ∈ x.orbit ∧ b ∈ y.orbit) ∧ a + b = 0) ↔ x = -y := by
+  simp_rw [mem_orbit]
+  constructor
+  · rintro ⟨a, b, ⟨rfl, rfl⟩, h⟩
+    rw [← mk_neg, eq, add_eq_zero_iff_eq_neg.mp h]
+  · rintro rfl
+    refine' ⟨-y.out, y.out, _⟩
+    simp_rw [mk_neg, out_eq, neg_add_self]
+#align gal_conj_classes.exist_mem_orbit_add_eq_zero GalConjClasses.exist_mem_orbit_add_eq_zero
+
+variable [IsSeparable F E]
+
+noncomputable nonrec def minpoly : GalConjClasses F E → F[X] :=
+  Quotient.lift (minpoly F) fun (a b : E) ⟨f, h⟩ =>
+    minpoly.eq_of_algHom_eq f.symm.toAlgHom f.symm.injective (IsSeparable.isIntegral F a)
+      (h ▸ (f.symm_apply_apply b).symm)
```

Comment:
This seems to be the only place `minpoly.eq_of_algHom_eq` is used. Should be easy to use [minpoly.minpoly_algHom](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/Minpoly/Basic.html#minpoly.minpoly_algHom) instead?